FramePack F1 with Video Input (Video Extension)

pftq opened the pull request:
I mirrored chaojie's approach to adding video input to SkyReels V2 (encoding the input video to latents for the video generation to build on) and did the same for Framepack here.  Run "python demo_gradio_f1_video.py" to try it.

ali0une commented:
Hi there!

Looks great but i'm stuck at :
`Encoding video frames:   0%|                | 0/7 [00:00<?, ?it/s]`

Nothing happens ...

Nvidia 3090 24Go VRAM.

pftq commented:
Are you on Windows? I've only tested on Linux so far.  Will check Windows a bit - but the Windows release is about 3 weeks old, so there may be some conflicts.

pftq commented:
Try now with updated demo_gradio_f1_video.py file: https://github.com/pftq/FramePack_with_Video_Input/blob/main/demo_gradio_f1_video.py

I reduced the VRAM use for the video encoding.

ali0une commented:
i can be so dumb sometimes ... i let the 'use teacache' checkbox checked, but i don't have teacache. :-/

Seems to work now ...

pftq commented:
Ah I've been habitually unchecking that.  Let me set that unchecked by default just in case.

ali0une commented:
man i was coding exactly [this](https://github.com/lllyasviel/FramePack/pull/491/commits/46d21990a447e44bce0d7f555ac5e51c01c116ba)! XD

user20212017 commented:
Is there a Youtube video guide? I need visual tutorial, to do it correctly ^^

pftq commented:
If you're using the Windows 7z, you need to rename the demo_gradio_f1_video.py to demo_gradio.py to overwrite the file that's there, since the run.bat will only run the original py file (demo_gradio.py).

> I had a working framepack install on windows, added the *_video.py to the original folder, pip installed decord, verified the install, modified the original run.bat to reference the _video.py file instead of the original framepack one. It launches the UI but when I try and add a video it is only accepting image extensions, when I change to All extensions and select a video then it just loads a broken image icon and throws: Exception in callback _ProactorBasePipeTransport._call_connection_lost(None) handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)> Traceback (most recent call last): File "asyncio\events.py", line 80, in _run File "asyncio\proactor_events.py", line 162, in _call_connection_lost

frosty3907 commented:
> If you're using the Windows 7z, you need to rename the demo_gradio_f1_video.py to demo_gradio.py to overwrite the file that's there, since the run.bat will only run the original py file (demo_gradio.py).
> 
> > I had a working framepack install on windows, added the *_video.py to the original folder, pip installed decord, verified the install, modified the original run.bat to reference the _video.py file instead of the original framepack one. It launches the UI but when I try and add a video it is only accepting image extensions, when I change to All extensions and select a video then it just loads a broken image icon and throws: Exception in callback _ProactorBasePipeTransport._call_connection_lost(None) handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)> Traceback (most recent call last): File "asyncio\events.py", line 80, in _run File "asyncio\proactor_events.py", line 162, in _call_connection_lost

Yep I already adjusted the .bat file to point to that one

pftq commented:
Did you update the Windows project to latest github code as well? Note the 7z is 3 weeks old.  You can also replace the diffusers_helper folder if the update bat is producing issues.

frosty3907 commented:
I did run update.bat which made a small update to something, I forgot what exactly

pftq commented:
Try just replacing the entire diffusers_helper - I am on Windows myself so replacing the diffusers_helper and renaming the py file (or changing what the run.bat points to) is what I did to have it working from the 7z.

ninehole commented:
Hello! Thanks for your program!
I can currently run it normally under Windows 10.
RTX 30360 12G, system memory 64G

A little bit not perfect is that the input video will lost too many frames.
Causing the final output video like a "frame-cut effect" in previous video, until it play to the new generated part  ,it become normal.
I guess this is designed to reduce GPU load ?
Or can it combine the previous video with full frame?

ninehole commented:
By the way, have you considered integrating "demo_gradio_f1" and "demo_gradio_video" into the same GUI?
For example, after the user generates a video from a pic , the user can do video extension in the same GUI ?
I think that will be great!
And maybe some code steps can be optimized by combining them together?

pftq commented:
I hotfixed it for now so it splices the original input video to the start and blends into the new generated footage.  I'm not sure why the VAE encoding sometimes corrupts the initial frames of the input video but I'll look into it later.

I also added a batch mode so you can generate multiple videos with different seeds, and the script now cleans up old partially finished videos so you just have the latest most complete video in the outputs folder.

ninehole commented:
Thank you so much for the update!
The version a few minutes ago cannot adjust VAE,
but I try it and the imput videos had no longer frame-dropping.
Now the finished video looks smooth!

I will try the newer version later, but I am not sure how to tune VAE in what situations?
Anyway, I am very satisfied with the result now, thank you!!!

pftq commented:
The VAE batch size option should be 2nd to bottom.  It should not be necessary to change it in most cases unless you see frame dropping again in the input video.

innokria commented:
I tried  your  skywork fix that didnt work may be due to my  hardware issue .. but this framepack was pretty light weight and I managed to run (with VRAM flag to false ) just to test the code . Whoow its working **the lady was about to fly** 

https://github.com/user-attachments/assets/6abeedcf-7493-4826-9e87-e512199686db

innokria commented:
Link to try out

https://huggingface.co/spaces/rahul7star/FramePack-F1

**Video 1 was generated with the prompt**
'A woman in a leather jacket and sunglasses riding a vintage motorcycle through a desert highway at sunset, her hair blowing wildly in the wind as the motorcycle kicks up dust, with the golden sun casting long shadows across the barren landscape.'

And then extended using
 **'A woman flies into space'**

pftq commented:
I've optimized the "Force Original Video Resolution (No Resizing)" a bit more so it uses less VRAM and doesn't OOM. It can handle really high resolution videos now and the detail is quite good, just takes long.  960x960 video with 25 steps takes about 10 min per second on H100, 1 hr per second on RTX 3090.

Below is a 1-second 1920x1088 video extended out to 2 seconds with 25 steps.  It takes about 30 min per second on a H100 and 2 hours per second at 100 steps, so Ill try to upload a longer extension later.
https://github.com/user-attachments/assets/e75b9ee7-793e-4fc8-b207-1ab2e024a00f

pftq commented:
Prompt: Woman with black hair and green eyes standing on a ship at sea. It is stormy and windy.
Distilled CFG: 1 (disabling it)
CFG: 3
Steps 25
VAE Batch: 128

pftq commented:
- Added prompt saving to the MP4 comments metadata.  Requires one additional dependency ("pip install imageio_ffmpeg")
- Exposed the Latent Window Size slider - this reduces the deterioration over time (because you do more frames per chunk) but also seems to further smooth out the transition between input video and new frames, probably because there are more frames to work with.

Old:
https://github.com/user-attachments/assets/e75b9ee7-793e-4fc8-b207-1ab2e024a00f

New with 18 latent window size instead of the default 9:
https://github.com/user-attachments/assets/827999a6-62e3-4b90-bc6e-1e3253d287d6

It's very subtle but the artifacts and slight color dip in the middle of the clip is smoothed out.

youzhe0305 commented:
Hello, I really appreciate your improvement of this project. Because of my personal requirement, I would like to add SDEdit functionality to this project. If I can do it well, I will also be willing to PR and share my modification (though my coding ability is not really good). Thus, if you can provide some quick tutorial about where to modify (which part of the code), it will help me a lot and accelerate my developing. Thanks for your contribution!!!

pftq commented:
> Hello, I really appreciate your improvement of this project. Because of my personal requirement, I would like to add SDEdit functionality to this project. If I can do it well, I will also be willing to PR and share my modification (though my coding ability is not really good). Thus, if you can provide some quick tutorial about where to modify (which part of the code), it will help me a lot and accelerate my developing. Thanks for your contribution!!!

Sorry, I don't even know what SDEdit is.  It took me several days full time to figure out the code for this edit, so I'm not sure I would be able to make any quick tutorials.

youzhe0305 commented:
> > Hello, I really appreciate your improvement of this project. Because of my personal requirement, I would like to add SDEdit functionality to this project. If I can do it well, I will also be willing to PR and share my modification (though my coding ability is not really good). Thus, if you can provide some quick tutorial about where to modify (which part of the code), it will help me a lot and accelerate my developing. Thanks for your contribution!!!
> 
> Sorry, it took me several days full time to figure out the code for this edit, so I'm not sure there's anything to make a quick tutorial of (it's just a lot of testing and going through the code). I also don't know what SDEdit is.

I see, and still thanks for you contribution!

pftq commented:
I added endframe support on top of the existing video input.  However, for now it only works with the original backward generation model - I haven't quite been able to get it working with F1 so far but maybe that changes later.
![image](https://github.com/user-attachments/assets/aaf6291a-531f-4f02-8232-d1214dc4f647)

innokria commented:
> I added end frame support on top of the video input (video extension). So now you can extend an existing video while maintaining the motion and also guide it to a specific end frame in mind.
> 
> For now it only works with the original backward generation model - I haven't quite been able to get it working with F1 so far but maybe that changes later. The F1 is a bit better at maintaining motion from the input video, so if you aren't using the end frame, I would stick with that version.
> 
> You can try it by downloading the latest version of this fork here: https://github.com/pftq/FramePack_with_Video_Input/ or git clone https://github.com/pftq/FramePack_with_Video_Input (pull instead of clone to update if you already downloaded FramePack) (pip install decord and pip install imageio_ffmpeg if you haven't already)
> 
> Then run "python demo_gradio_video+endframe.py" (different file from the first demo_gradio_f1_video.py to not mess up the f1 version).
> 
> If the end frame is very different from the input video, you might need to reduce context frames to avoid jumpcutting - context frames makes it adhere more to the video input, so it might not know what to do if the end frame looks too different. I recommend trying it reduced to 2 anyway to start as it generally allows more "freedom" for the video to deviate from the video and raising it only if it looks too inconsistent. ![image](https://private-user-images.githubusercontent.com/3508096/442704061-aaf6291a-531f-4f02-8232-d1214dc4f647.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwNzY4NzksIm5iZiI6MTc0NzA3NjU3OSwicGF0aCI6Ii8zNTA4MDk2LzQ0MjcwNDA2MS1hYWY2MjkxYS01MzFmLTRmMDItODIzMi1kMTIxNGRjNGY2NDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTJUMTkwMjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGU3NTNiMDRjZDFjYzQzMTk2YzEzNjk1NzM2ZDRlYzU5MmUzMzg2ZWViMmZmNGVhYTc5Mjc5NmE2MGJkZTcxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Usa2Fkio-LhdOkyqzKSzxC8Wzz2bnmSmGyjmZWao5x8)

I am going to try this now , what is the use case of prompt ? can i add a prompt and last frame image to create a video ?

pftq commented:
> > I added end frame support on top of the video input (video extension). So now you can extend an existing video while maintaining the motion and also guide it to a specific end frame in mind.
> > For now it only works with the original backward generation model - I haven't quite been able to get it working with F1 so far but maybe that changes later. The F1 is a bit better at maintaining motion from the input video, so if you aren't using the end frame, I would stick with that version.
> > You can try it by downloading the latest version of this fork here: https://github.com/pftq/FramePack_with_Video_Input/ or git clone https://github.com/pftq/FramePack_with_Video_Input (pull instead of clone to update if you already downloaded FramePack) (pip install decord and pip install imageio_ffmpeg if you haven't already)
> > Then run "python demo_gradio_video+endframe.py" (different file from the first demo_gradio_f1_video.py to not mess up the f1 version).
> > If the end frame is very different from the input video, you might need to reduce context frames to avoid jumpcutting - context frames makes it adhere more to the video input, so it might not know what to do if the end frame looks too different. I recommend trying it reduced to 2 anyway to start as it generally allows more "freedom" for the video to deviate from the video and raising it only if it looks too inconsistent. ![image](https://private-user-images.githubusercontent.com/3508096/442704061-aaf6291a-531f-4f02-8232-d1214dc4f647.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwNzY4NzksIm5iZiI6MTc0NzA3NjU3OSwicGF0aCI6Ii8zNTA4MDk2LzQ0MjcwNDA2MS1hYWY2MjkxYS01MzFmLTRmMDItODIzMi1kMTIxNGRjNGY2NDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTJUMTkwMjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGU3NTNiMDRjZDFjYzQzMTk2YzEzNjk1NzM2ZDRlYzU5MmUzMzg2ZWViMmZmNGVhYTc5Mjc5NmE2MGJkZTcxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Usa2Fkio-LhdOkyqzKSzxC8Wzz2bnmSmGyjmZWao5x8)
> 
> I am going to try this now , what is the use case of prompt ? can i add a prompt and last frame image to create a video ?
> 
> I tried I am getting
> 
> ![image](https://private-user-images.githubusercontent.com/1271878/443865978-ec7418cf-6ce1-46ec-afe2-a6ac980c903d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyNzAyNTQsIm5iZiI6MTc0NzI2OTk1NCwicGF0aCI6Ii8xMjcxODc4LzQ0Mzg2NTk3OC1lYzc0MThjZi02Y2UxLTQ2ZWMtYWZlMi1hNmFjOTgwYzkwM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTVUMDA0NTU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzEzZThjNTZjYmQ0ZjY1OTE5MWQxMGYzN2I3MWNkOTdmMzUwMjYyNWJjZTEwYjVjZTg2OTNkZTYzMDRmYTJhMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gQzDeO6y90EH54vdW0vD2XgA066PYboudHNuLMOhQ_k)
> 
> File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 158, in **call** query = apply_rotary_emb_transposed(query, image_rotary_emb) File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 103, in apply_rotary_emb_transposed out = x.float() * cos + x_rotated.float() * sin **RuntimeError: The size of tensor a (15780) must match the size of tensor b (17300) at non-singleton dimension 1**
> 
> Link https://huggingface.co/spaces/rahul7star/FramePack-video-endframe
> 
> ChatGPT has couple of fixes for the method **apply_rotary_emb_transposed** but keen to you have you got this error

A mismatch would usually either be caused by a frame count or width/height. I'd need to know more details about the video and the settings you're using on the GUI to know what the issue is.

innokria commented:
> > > I added end frame support on top of the video input (video extension). So now you can extend an existing video while maintaining the motion and also guide it to a specific end frame in mind.
> > > For now it only works with the original backward generation model - I haven't quite been able to get it working with F1 so far but maybe that changes later. The F1 is a bit better at maintaining motion from the input video, so if you aren't using the end frame, I would stick with that version.
> > > You can try it by downloading the latest version of this fork here: https://github.com/pftq/FramePack_with_Video_Input/ or git clone https://github.com/pftq/FramePack_with_Video_Input (pull instead of clone to update if you already downloaded FramePack) (pip install decord and pip install imageio_ffmpeg if you haven't already)
> > > Then run "python demo_gradio_video+endframe.py" (different file from the first demo_gradio_f1_video.py to not mess up the f1 version).
> > > If the end frame is very different from the input video, you might need to reduce context frames to avoid jumpcutting - context frames makes it adhere more to the video input, so it might not know what to do if the end frame looks too different. I recommend trying it reduced to 2 anyway to start as it generally allows more "freedom" for the video to deviate from the video and raising it only if it looks too inconsistent. ![image](https://private-user-images.githubusercontent.com/3508096/442704061-aaf6291a-531f-4f02-8232-d1214dc4f647.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwNzY4NzksIm5iZiI6MTc0NzA3NjU3OSwicGF0aCI6Ii8zNTA4MDk2LzQ0MjcwNDA2MS1hYWY2MjkxYS01MzFmLTRmMDItODIzMi1kMTIxNGRjNGY2NDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTJUMTkwMjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGU3NTNiMDRjZDFjYzQzMTk2YzEzNjk1NzM2ZDRlYzU5MmUzMzg2ZWViMmZmNGVhYTc5Mjc5NmE2MGJkZTcxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Usa2Fkio-LhdOkyqzKSzxC8Wzz2bnmSmGyjmZWao5x8)
> > 
> > 
> > I am going to try this now , what is the use case of prompt ? can i add a prompt and last frame image to create a video ?
> > I tried I am getting
> > ![image](https://private-user-images.githubusercontent.com/1271878/443865978-ec7418cf-6ce1-46ec-afe2-a6ac980c903d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyNzAyNTQsIm5iZiI6MTc0NzI2OTk1NCwicGF0aCI6Ii8xMjcxODc4LzQ0Mzg2NTk3OC1lYzc0MThjZi02Y2UxLTQ2ZWMtYWZlMi1hNmFjOTgwYzkwM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTVUMDA0NTU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzEzZThjNTZjYmQ0ZjY1OTE5MWQxMGYzN2I3MWNkOTdmMzUwMjYyNWJjZTEwYjVjZTg2OTNkZTYzMDRmYTJhMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gQzDeO6y90EH54vdW0vD2XgA066PYboudHNuLMOhQ_k)
> > File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 158, in **call** query = apply_rotary_emb_transposed(query, image_rotary_emb) File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 103, in apply_rotary_emb_transposed out = x.float() * cos + x_rotated.float() * sin **RuntimeError: The size of tensor a (15780) must match the size of tensor b (17300) at non-singleton dimension 1**
> > Link https://huggingface.co/spaces/rahul7star/FramePack-video-endframe
> > ChatGPT has couple of fixes for the method **apply_rotary_emb_transposed** but keen to you have you got this error
> 
> A mismatch would usually either be caused by a frame count or width/height. I'd need to know more details about the video and the settings you're using on the GUI to know what the issue is.

Ok I will simulate again ,BTW what's the use case of **prompt text**  here

pftq commented:
I don't understand the question - it's the same way you would use prompt text for any other video generation.

innokria commented:
> > > > I added end frame support on top of the video input (video extension). So now you can extend an existing video while maintaining the motion and also guide it to a specific end frame in mind.
> > > > For now it only works with the original backward generation model - I haven't quite been able to get it working with F1 so far but maybe that changes later. The F1 is a bit better at maintaining motion from the input video, so if you aren't using the end frame, I would stick with that version.
> > > > You can try it by downloading the latest version of this fork here: https://github.com/pftq/FramePack_with_Video_Input/ or git clone https://github.com/pftq/FramePack_with_Video_Input (pull instead of clone to update if you already downloaded FramePack) (pip install decord and pip install imageio_ffmpeg if you haven't already)
> > > > Then run "python demo_gradio_video+endframe.py" (different file from the first demo_gradio_f1_video.py to not mess up the f1 version).
> > > > If the end frame is very different from the input video, you might need to reduce context frames to avoid jumpcutting - context frames makes it adhere more to the video input, so it might not know what to do if the end frame looks too different. I recommend trying it reduced to 2 anyway to start as it generally allows more "freedom" for the video to deviate from the video and raising it only if it looks too inconsistent. ![image](https://private-user-images.githubusercontent.com/3508096/442704061-aaf6291a-531f-4f02-8232-d1214dc4f647.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwNzY4NzksIm5iZiI6MTc0NzA3NjU3OSwicGF0aCI6Ii8zNTA4MDk2LzQ0MjcwNDA2MS1hYWY2MjkxYS01MzFmLTRmMDItODIzMi1kMTIxNGRjNGY2NDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTJUMTkwMjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGU3NTNiMDRjZDFjYzQzMTk2YzEzNjk1NzM2ZDRlYzU5MmUzMzg2ZWViMmZmNGVhYTc5Mjc5NmE2MGJkZTcxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Usa2Fkio-LhdOkyqzKSzxC8Wzz2bnmSmGyjmZWao5x8)
> > > 
> > > 
> > > I am going to try this now , what is the use case of prompt ? can i add a prompt and last frame image to create a video ?
> > > I tried I am getting
> > > ![image](https://private-user-images.githubusercontent.com/1271878/443865978-ec7418cf-6ce1-46ec-afe2-a6ac980c903d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyNzAyNTQsIm5iZiI6MTc0NzI2OTk1NCwicGF0aCI6Ii8xMjcxODc4LzQ0Mzg2NTk3OC1lYzc0MThjZi02Y2UxLTQ2ZWMtYWZlMi1hNmFjOTgwYzkwM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTVUMDA0NTU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzEzZThjNTZjYmQ0ZjY1OTE5MWQxMGYzN2I3MWNkOTdmMzUwMjYyNWJjZTEwYjVjZTg2OTNkZTYzMDRmYTJhMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gQzDeO6y90EH54vdW0vD2XgA066PYboudHNuLMOhQ_k)
> > > File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 158, in **call** query = apply_rotary_emb_transposed(query, image_rotary_emb) File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 103, in apply_rotary_emb_transposed out = x.float() * cos + x_rotated.float() * sin **RuntimeError: The size of tensor a (15780) must match the size of tensor b (17300) at non-singleton dimension 1**
> > > Link https://huggingface.co/spaces/rahul7star/FramePack-video-endframe
> > > ChatGPT has couple of fixes for the method **apply_rotary_emb_transposed** but keen to you have you got this error
> > 
> > 
> > A mismatch would usually either be caused by a frame count or width/height. I'd need to know more details about the video and the settings you're using on the GUI to know what the issue is.
> 
> Ok I will simulate again ,BTW what's the use case of **prompt text** here

All good , the default settings are working well , earlier i messed up the config .  
Thanks its all working well with default settings

innokria commented:
Audio addition to these videos will be amazing any thoughts ?

matt-bendel commented:
> > > > > I added end frame support on top of the video input (video extension). So now you can extend an existing video while maintaining the motion and also guide it to a specific end frame in mind.
> > > > > For now it only works with the original backward generation model - I haven't quite been able to get it working with F1 so far but maybe that changes later. The F1 is a bit better at maintaining motion from the input video, so if you aren't using the end frame, I would stick with that version.
> > > > > You can try it by downloading the latest version of this fork here: https://github.com/pftq/FramePack_with_Video_Input/ or git clone https://github.com/pftq/FramePack_with_Video_Input (pull instead of clone to update if you already downloaded FramePack) (pip install decord and pip install imageio_ffmpeg if you haven't already)
> > > > > Then run "python demo_gradio_video+endframe.py" (different file from the first demo_gradio_f1_video.py to not mess up the f1 version).
> > > > > If the end frame is very different from the input video, you might need to reduce context frames to avoid jumpcutting - context frames makes it adhere more to the video input, so it might not know what to do if the end frame looks too different. I recommend trying it reduced to 2 anyway to start as it generally allows more "freedom" for the video to deviate from the video and raising it only if it looks too inconsistent. ![image](https://private-user-images.githubusercontent.com/3508096/442704061-aaf6291a-531f-4f02-8232-d1214dc4f647.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwNzY4NzksIm5iZiI6MTc0NzA3NjU3OSwicGF0aCI6Ii8zNTA4MDk2LzQ0MjcwNDA2MS1hYWY2MjkxYS01MzFmLTRmMDItODIzMi1kMTIxNGRjNGY2NDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTJUMTkwMjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGU3NTNiMDRjZDFjYzQzMTk2YzEzNjk1NzM2ZDRlYzU5MmUzMzg2ZWViMmZmNGVhYTc5Mjc5NmE2MGJkZTcxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Usa2Fkio-LhdOkyqzKSzxC8Wzz2bnmSmGyjmZWao5x8)
> > > > 
> > > > 
> > > > I am going to try this now , what is the use case of prompt ? can i add a prompt and last frame image to create a video ?
> > > > I tried I am getting
> > > > ![image](https://private-user-images.githubusercontent.com/1271878/443865978-ec7418cf-6ce1-46ec-afe2-a6ac980c903d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyNzAyNTQsIm5iZiI6MTc0NzI2OTk1NCwicGF0aCI6Ii8xMjcxODc4LzQ0Mzg2NTk3OC1lYzc0MThjZi02Y2UxLTQ2ZWMtYWZlMi1hNmFjOTgwYzkwM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MTVUMDA0NTU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzEzZThjNTZjYmQ0ZjY1OTE5MWQxMGYzN2I3MWNkOTdmMzUwMjYyNWJjZTEwYjVjZTg2OTNkZTYzMDRmYTJhMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gQzDeO6y90EH54vdW0vD2XgA066PYboudHNuLMOhQ_k)
> > > > File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 158, in **call** query = apply_rotary_emb_transposed(query, image_rotary_emb) File "/home/user/app/diffusers_helper/models/hunyuan_video_packed.py", line 103, in apply_rotary_emb_transposed out = x.float() * cos + x_rotated.float() * sin **RuntimeError: The size of tensor a (15780) must match the size of tensor b (17300) at non-singleton dimension 1**
> > > > Link https://huggingface.co/spaces/rahul7star/FramePack-video-endframe
> > > > ChatGPT has couple of fixes for the method **apply_rotary_emb_transposed** but keen to you have you got this error
> > > 
> > > 
> > > A mismatch would usually either be caused by a frame count or width/height. I'd need to know more details about the video and the settings you're using on the GUI to know what the issue is.
> > 
> > 
> > Ok I will simulate again ,BTW what's the use case of **prompt text** here
> 
> All good , the default settings are working well , earlier i messed up the config . Thanks its all working well with default settings
> 
> **Sequence**
> 
> * Penguin was walking
> * Extended ( Penguin is flying )
> * last frame ( penguin is back to original location)
> 
>  2025-05-15_03-10-39_framepack-videoinput-endframe_608-5sec_seed-31337_steps-25_distilled-3_cfg-1_final.mp4

I am having the same issue… what was wrong with your config?

pftq commented:
If you can share the duration and width/height of the input video, I can double check for edge cases - sometimes it's because of a video being not divisible by the number of frames being generated or something to that effect.

pftq commented:
Try a video longer than 1 second.  Because Hunyuan (the model) generates in 1 second chunks, it needs more duration to know what to continue.

matt-bendel-descript commented:
> Try a video longer than 1 second. Because Hunyuan (the model) generates in 1 second chunks, it needs more duration to know what motion from the video to continue.

That did indeed do the trick. Going to 1.5s solved the issue. Curiously, with the same 1s video input I had no issues with the video-only input demo (demo_gradio_f1_video.py). Any thoughts as to why that may be? One other question: could this be easily extend to both a start and end video?

pftq commented:
I had extra code to handle the 1-sec video on the f1.  I'll add some similar code to handle the 1-sec case on the non-f1 version later, generally speaking it doesn't do as well without more context so it's sort of a bandaid fix.

sooxt98 commented:
the existing video become ghosting after extend

pftq commented:
Try increasing the VAE batch size (if you have enough VRAM) - that is what affects the quality of the input video and its motion.

bigmanong commented:
I git pulled this into an existing windows install of framepack F1, got it up and running by installing decord, but ran into the issue with the gradio still only allowing image inputs.  I then replaced the diffusers_helper folder, but now when I run it, I get an error saying decord is not found, but I can verify it is there.  

I am decent at following instructions, but have very little actual knowledge of python.  Can anyone help me figure out what I did here?

pftq commented:
Do a fresh clone instead of a pull:
git clone https://github.com/pftq/FramePack_with_Video_Input

jimmaybones commented:
I'm having no luck installing the extension. I really want to test the video to video option. I have FramePack-F-1 installed and I have followed the instructions, but I'm not getting anything to work. It keeps going back to the standard FramePack-F-1 page with only the image to video available. Can you please write a step-by-step guide for people like me who have literally opened a cmd prompt a handful of times.

pftq commented:
> I'm having no luck installing the extension. I really want to test the video to video option. I have FramePack-F-1 installed and I have followed the instructions, but I'm not getting anything to work. It keeps going back to the standard FramePack-F-1 page with only the image to video available. Can you please write a step-by-step guide for people like me who have literally opened a cmd prompt a handful of times.

I need to know what operating system you are using and what steps you took so far.

jimmaybones commented:
I'm on windows ten. I installed FramePack and then installed FramePack-F-1. Then I downloaded the FramePack with video input extension. I tried running python demo_gradio_f1_video.py, but it said there wasn't a gradio. I moved the files over to the FramePack-F-1 extension, then tried running again and got the same error. I replaced the diffusion helper folder and that still didn't help. Is there a way I can start from the beginning with very simple and straightforward steps(I have absolutely no coding experience)? thanks

pftq commented:
Did you do this step for Windows (from the main post)? I ask because the step suggests replacing the demo_gradio.py file and not running demo_gradio_f1_video.py separately. Also you should confirm that FramePack itself (original) works on your Windows setup before adding this.

`If you're using the Windows 7z, you need to also replace the diffusers_helper folder since the 7z is 3 weeks behind and also rename the demo_gradio_f1_video.py to demo_gradio.py to overwrite the file that's there, since the run.bat will only run the original py file (demo_gradio.py). You still need the decord + imageio above, which should be installed in the system/python folder via:

python -m pip install decord==0.6.0
python -m pip install imageio_ffmpeg==0.6.0`

jimmaybones commented:
Yeah I installed the decord and imageio_ffmpeg and renamed the gradio. I still get the FramePack-F-1 version coming up with image to video.

pftq commented:
The F-1 version shouldn't exist if you overwrote it with the renamed demo_gradio_f1_video.py.  Maybe double check that step.  Also just to confirm, you are still using the run.bat file to launch?

pftq commented:
It's renaming demo_gradio_f1_video.py to demo_gradio.py - it should be overwriting the demo file that is already existing.

> Thanks for the quick responses and help.
> 
> Yes, confirming the use of run.bat to launch. I was taking the extension files and having those overwrite the original. There was not a run.bat in the extension. I'm starting to think I should start from scratch. This would be my methodology:
> 
> Get FramePack_F-1 Download the extension files Replace the files in F-1 Rename the gradio_f1_video.py to gradio_video.py Should work right?

pftq commented:
The default guidance is much lower than the 10 in the original but the prompt does have an effect.  Context Frames also can be reduced to 2 to give more weight to the prompt.  Both of these sacrifice maintaining detail from the input video though, so you have to pick your priority.

Fabrice-TIERCELIN commented:
Hi @pftq,

The issue is also here with an end frame. My code is the same as this PR. I have only added a randomized seed. Here are the parameters:


                    "./img_examples/Example1.mp4", # input_video
                    "./img_examples/Example1.png", # end_frame
                    0.0, # end_frame_weight
                    "View of the sea as far as the eye can see, from the seaside, a piece of land is barely visible on the horizon at the middle, the sky is radiant, reflections of the sun in the water, photorealistic, realistic, intricate details, 8k, insanely detailed",
                    "Missing arm, unrealistic position, blurred, blurry", # n_prompt
                    True, # randomize_seed
                    42, # seed
                    1, # batch
                    640, # resolution
                    1, # total_second_length
                    9, # latent_window_size
                    25, # steps
                    1.0, # cfg
                    10.0, # gs
                    0.0, # rs
                    6, # gpu_memory_preservation
                    False, # use_teacache
                    False, # no_resize
                    16, # mp4_crf
                    5, # num_clean_frames
                    128, # default_vae

Can you please test on your side with my inputs?

https://huggingface.co/spaces/Fabrice-TIERCELIN/SUPIR/resolve/main/img_examples/Example1.png?download=true

https://huggingface.co/spaces/Fabrice-TIERCELIN/SUPIR/resolve/main/img_examples/Example1.mp4?download=true

pftq commented:
I noticed you have total length at 1 second.  Can you try the default value (5)? The input video and end frame each require a separate latent to guide the video, so it could be that you just don't have enough runway for the generation.

Fabrice-TIERCELIN commented:
OK, I will test it. Meanwhile, can you proofread my improvement suggestions?

j2l commented:
I stopped trying.

pftq commented:
That is outside the scope of the fork - the fork just uses the default 640 from the main branch.  There is no option to change the resolution, so I'm not sure what you are addressing there.  If you are addressing the "force original resolution" option, that is more of a "use-at-your-own-risk" that directly uses the original dimensions, which doesn't really result in any error but at worse some degradation in generation quality (if any, I haven't seen any in my own tests).

Fabrice-TIERCELIN commented:
Your fork is the only place that adds the resolution component in the interface. I would be out-of-scope to write it elsewhere. But yes, it's not critical as the component is hidden, so the component is hard to change. Anyway, people knows how to fix the problem now.

pftq commented:
Yes, it's hidden because it's not meant to be changed.  It just organizes the hardcoded numbers in one place to make it easier to follow (640 is hardcoded as the resolution in the main branch).

Fabrice-TIERCELIN commented:
Maybe I have found the rootcause of the jumpcut for end frame. After each chunk of generation, we call `soft_append_bcthw()` to join the parts. We should call it twice for the last part:

1. To join with the previous generation
2. And to join with the original video

What do you think about it, @pftq?

Fabrice-TIERCELIN commented:
> Is there a way to download this while keeping the original functions as well? As in I want to be able to use original frame pack to generate a video then use this to extend it. Looking through the posts it seems like I need to overwrite demo_gradio_f1_video.py to demo_gradio.py and I am not sure if that will break something. I am new to this and trying to figure out how this works. Can I clone use the git clone https://github.com/pftq/FramePack_with_Video_Input to get a 2nd version to edit without affecting the original? Also a video tutorial would be helpful if anyone knows a good one. Thanks!

I have created a merged version on [HuggingFace](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack). You can see the code clicking on [Files](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/tree/main). You have to overwrite the files:

1. [app.py](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/blob/main/app.py)
2. [requirements.txt](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/blob/main/requirements.txt)
3. [bucket_tools.py](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/blob/main/diffusers_helper/bucket_tools.py)
4. [Example*](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/blob/main/img_examples/Example1.mp4)

You also have to reuse the version of Gradio from [README.md](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/blob/main/README.md). As my version is running on HuggingFace, you will have to do some adaptations like removing the use of `spaces` in [app.py](https://huggingface.co/spaces/Fabrice-TIERCELIN/FramePack/blob/main/app.py). Coding knowledge would be better. Good luck.

ryukbk commented:
Awesome project. How about introducing this function into a more actively maintained branch of Framepack such as https://github.com/git-ai-code/FramePack-eichi which has nice functions like working lora support?

pftq commented:
There have been requests for a bunch of different forks already, so feel free to send or recommend it to the other authors.  As long as I'm credited for the original code, I don't mind if it gets copied into other projects.

Fabrice-TIERCELIN commented:
This PR feature is at last [available online](https://huggingface.co/spaces/makululinux/FramePack-F1) 😃 (among other new features)

andreyshade commented:
Is it possible to set an image as a start frame not a video file?

Fabrice-TIERCELIN commented:
- If you want end frame, you can't here. You should rather use [Add Start and End Frame control，works great!](https://github.com/lllyasviel/FramePack/pull/167)
- If you don't want end frame, just use the standard version

pftq commented:
Grok is pretty good for stuff like this: https://x.com/i/grok/share/sEA8m8OOwyy2so5K0OasbYVlM
```
The issue you're encountering is likely because the decord package was installed into your system's global Python environment, but the FramePack one-click Windows package uses a bundled Python interpreter (located in a subdirectory like framepack_cu126_torch26\system\python or similar—confirm the exact path in your unpacked folder structure).

To fix this:Navigate to the folder containing the bundled python.exe (e.g., open CMD and cd C:\AI_Video_Gen\framepack_cu126_torch26\system\python if that's the path; adjust based on your directory).
Run the installation commands using that specific Python:

python -m pip install decord==0.6.0

python -m pip install imageio_ffmpeg==0.6.0

If you're running demo_gradio_video+endframe.py directly (instead of via run.bat), use the bundled Python to execute it:

python demo_gradio_video+endframe.py

(If you're not already in the webui folder, navigate back there first and use the full path to the bundled python.exe, e.g., ..\..\system\python\python.exe demo_gradio_video+endframe.py.)

```
